MacBry-Maciej-Bryja/SimpleTodoApp1/src/main/java/com/mac/bry/simpleTodo/appController/MainController.java

```diff
@@ -0,0 +1,112 @@
+package com.mac.bry.simpleTodo.appController;
+
+import java.nio.charset.Charset;
+import java.time.LocalDate;
+import java.util.Random;
+
+import org.apache.log4j.Logger;
+
+import com.mac.bry.simpleTodo.DAO.UserDAO;
+import com.mac.bry.simpleTodo.Enums.LoginOption;
+import com.mac.bry.simpleTodo.entity.User;
+import com.mac.bry.simpleTodo.utilitis.DataReader;
+import com.mac.bry.simpleTodo.utilitis.EmailSender;
+import com.mac.bry.simpleTodo.utilitis.PasswordUtillity;
+
+public class MainController {
+
+	private Logger logger = Logger.getLogger(MainController.class.getName());
+	
+	
+	private DataReader dataReader;
+	private UserDAO userDAO;
+	private MenuController menuController;
+	private User loggedUser;
+
+	public MainController() {
+		super();
+		this.dataReader = new DataReader();
+		this.userDAO = new UserDAO();
+		this.menuController = new MenuController();
+	}
+
+	public void programLoop() {
+		LoginOption option;
+		printLoginOptions();
+
+		while ((option = LoginOption.createFromInt(dataReader.readNumber())) != LoginOption.EXIT) {
+			switch (option) {
+			case LOGIN:
+				if (login()) {
+					;
+					menuController.menuLoop();
+				} else {
+					System.out.println("Try again");
+					login();
+				}
+				break;
+
+			case REGISTR:
+				registr();
+				break;
+
+			case PASSWORD_RESET:
+				resetPassword();
+				break;
+
+			default:
+				System.err.println("\nNo such option");
+				
+				String name = new Object(){}.getClass().getEnclosingMethod().getName();
+				logger.warn("[" + LocalDate.now().toString() + "] ---> Run " + name + " No such option");
+				
+				programLoop();
```

Comment:
Po co tutaj wywołanie rekurencyjne?

Reply:
Jeśli nie wybiorę opcji z enuma to chciałem zacząć od nowa pętle